Objects/genericaliasobject.c

```diff
@@ -11,8 +11,17 @@ typedef struct {
     PyObject *args;
     PyObject *parameters;
     PyObject* weakreflist;
+    // Whether we're a starred type, e.g. *tuple[int].
+    // Only supported for `tuple`.
+    int starred;
 } gaobject;
 
+typedef struct {
+    PyObject_HEAD
+    PyObject *obj;  /* Set to NULL when iterator is exhausted */
+    int exhausted;
```

Comment:
If we're setting `obj` to `NULL` only when it's exhausted, can't we check for that instead of checking for the `exhausted` flag? Then we can get rid of one field and save some memory. Or is there some situation where `obj` is `NULL` even when not exhausted?